### PR TITLE
Class Features Update

### PR DESCRIPTION
**Pull request by josephpatrickgreen**
## About The Pull Request
Adds d&d class features to a variety of classes. Also ports Lockpicking from Stonekeep.

## Amazon
Created and added a new trait "Bed Breaker" or "Death By Snoo-Snoo" in the code which doubles sex damage. This trait has been given to the amazon. This requires strong intent so that it is optional.

## Bard
Vicious Mockery as a spell added to bards which does the following:
-screams a randomly generated insult at the target to hurt their feelings.
-deals VERY LIGHT burn damage (sick burn dude)
-blinds for 1 second
-blurs vision for 15 seconds
-lowers fortune for 30 seconds and speed slightly

Example insults:
"Scamper home, you foppish beef-witted kobold-kisser!"
"Tremble, you yeasty miscreant codpiece!"
"Flounder, you spongy spleeny clack-dish!"
There are countless combinations.

## Barbarian
Rage as a spell added to barbarians which does the following:
-adds 9 stress to user for 10 minutes.
-adds 4 str, 2 spd, 2 end for 45 seconds.
-on buff end ZEROES your stamina and gives you sore muscles

Rage is meant to be either a last resort combat tactic but can be used to rushdown mobs or mine and chop down trees faster. Spamming rage is meant to result in a heart attack from the major stress.

Danger Sense: barbarians get perfect tracking

Death By Snoo-Snoo: Ravager subclass deals double sex damage

## Rogue
Thieves' Cant as a language added to rogues which is meant to give them a discreet way of speaking to eachother with emotes.

Lockpicking has been added

Dirty fighting (nut cracker trait) has been added to swashbucklers

## Misc
Werewolves also get death by snoo-snoo

## Why It's Good For The Game
Vicious Mockery:
It's just a fun D&D spell that can be added for bards and jesters. The randomly generated insults are fun to play around with. It isn't super harmful or OP. It allows for a quick retort for rp purposes as well as an escape or fighting edge to an otherwise non-combat oriented class. It's an overall improvement to the bard class and possibly jester class in the future.

Rage:
Barbarians are supposed to be scary on the warpath, their main feature is rage which is now here (yay). If I hear the warcry from a character dressed in hides I am RUNNING. That's fun for me.

Thieves' Cant:
Thieves have a subtle way of comunitcating in D&D where people within meters may not even know how a discussion is happening. This language uses gestures and everyday activity to denote words. Now you can communicate "We strike at first light" with a simple wink.

### CODE REVIEW

**Comment by josephpatrickgreen** [comment]
image

**Comment by Decoy1240** [comment]
Don’t add this until hamhole fixes good lover doing double sex damage and not his new trait

**Comment by josephpatrickgreen** [comment]
> Don’t add this until hamhole fixes good lover doing double sex damage and not his new trait

that does double pleasure. I'm not sure it's possible to have both traits unless you're a werewolf (whoever put that for werewolves is a real freak)

**Comment by josephpatrickgreen** [comment]
> Don’t add this until hamhole fixes good lover doing double sex damage and not his new trait

fixed

**Comment by EtheoBoxxman** [comment]
I dont think the knock spell is justifiable for rogues. Theyve got the best possible combination of stealth and pickpocketing, and there's already a lockpicking skill template in the code right now- Which reminds me, there's already lockpicking code in Azure Peak, Ive heard.

**Comment by josephpatrickgreen** [comment]
> also you bumped barbarians default con by [1](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExN3R5ODJpMW4zdmlkaDNqaWd5dmZ0OTN4bXpsNDZoamtxa25ydDh6ZiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/BtedgmzGNCiuk/giphy.gif)

it was requested by someone to not decrease ravager con by one so i put it in there for em, if its an issue I can remove it.

**Comment by josephpatrickgreen** [comment]
> remove knock and double check your DM checks this time around and I'll thumbs up it. only because lockpicking needs to be a thing not a spell. can either cherry pick it or make our own version soon. spell would be problematic for more than a few reasons.

everything compiles now and I've tested all the stuff I've done, had to learn dm (was my first attempt) I understand the syntax now.

**Comment by josephpatrickgreen** [comment]
> I dont think the knock spell is justifiable for rogues. Theyve got the best possible combination of stealth and pickpocketing, and there's already a lockpicking skill template in the code right now- Which reminds me, there's already lockpicking code in Azure Peak, Ive heard.

i cant find that repo but when I do I'll port it over. knock is removed now.

File: code/modules/spells/roguetown/barbarian.dm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	timer = 10 MINUTES
+	stressadd = 8
+	max_stacks = 10 //don't rage spam or you WILL have a heart attack
+
+//claws for the ravager
+/obj/effect/proc_holder/spell/self/rav_claws
+	name = "Ravager Claws"
+	desc = "!"
+	overlay_state = "claws"
+	antimagic_allowed = TRUE
```
[the remark this answers is not in the document]
Comment (josephpatrickgreen) [reply code/modules/spells/roguetown/barbarian.dm:59]: i thought this said it was allowed during antimagic effect.

[the remark this answers is not in the document]
Comment (josephpatrickgreen) [reply code/modules/spells/roguetown/barbarian.dm:59]: its set to true for vv

**Comment by Tree445** [comment]
image
needs an update due to merging another pr with this dm.

Also, you gave monks nofalldamage1, which is no fall damage at all. There is also a spell that temporarily removes fall damage.

**Comment by josephpatrickgreen** [comment]
> image needs an update due to merging another pr with this dm.
> 
> Also, you gave monks nofalldamage1, which is no fall damage at all. There is also a spell that temporarily removes fall damage.

merge finished, on the fall damage spell thing, good idea.

**Comment by josephpatrickgreen** [comment]
Removed the slowfall from monks

added lockpicks, lockpicking, lock pick rings

only rogues have lockpicks for now

**Comment by josephpatrickgreen** [comment]
alright, the skill has been given to some classes and the items to fewer classes. lockpicking now gives exp.

**Pull request merged**